### PR TITLE
DPP-1577 Remove Displayed Value When Funding Band Cap Exceeded

### PR DESCRIPTION
**Pull request by lazcool**
https://skillsfundingagency.atlassian.net/browse/DPP-1577
Originally branched off DPP-1000

### CODE REVIEW

File: src/SFA.DAS.CloudService/ServiceConfiguration.Local.cscfg [hunk 1]
```diff
@@ -13,7 +13,7 @@
       <Setting name="LoggingRedisConnectionString" value="" />
       <Setting name="LoggingRedisKey" value="" />
       <Setting name="TokenCertificateThumbprint" value="6B7ACC520305BFDB4F7252DAEB2177CC091FAAE1" />
-      <Setting name="MyaBaseUrl" value="https://dev-eas.apprenticeships.sfa.bis.gov.uk/" />
+      <Setting name="MyaBaseUrl" value="https://test2-eas.apprenticeships.sfa.bis.gov.uk/" />
```
Comment (scottywakefield) [inline src/SFA.DAS.CloudService/ServiceConfiguration.Local.cscfg]: 😮 you beat me to it.

File: src/SFA.DAS.EmployerApprenticeshipsService.Web.UnitTests/TestHelper.cs [hunk 2]
```diff
@@ -12,6 +13,12 @@ public static bool AreEqual(object obj1, object obj2)
             return result.AreEqual;
         }
 
+        public static bool EnumerablesAreEqual<T>(IEnumerable<T> expected, IEnumerable<T> actual)
```
Comment (scottywakefield) [inline src/SFA.DAS.EmployerApprenticeshipsService.Web.UnitTests/TestHelper.cs]: Another option to this is using FluentAssertions which has a `listA.Should().AllBeEquivalentTo(listB)` or similar. Suggestion only.

File: src/SFA.DAS.EmployerApprenticeshipsService.Web.UnitTests/ViewModels/WhenCreatedApprenticeshipListItemGroupViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+        public void AndNoApprenticeshipsThenThereAreNoApprenticeshipsOverFundingLimit()
+        {
+            var trainingProgram = new Framework();
+
+            var group = new ApprenticeshipListItemGroupViewModel(new ApprenticeshipListItemViewModel[0], trainingProgram);
+
+            Assert.AreEqual(0, group.ApprenticeshipsOverFundingLimit);
+        }
+
+        [TestCase("2020-1-15",   "99", "2020-1-10", "2020-1-20", 100, 0, Description = "StarDatet in band, cost less than cap")]
```
Comment (scottywakefield) [inline src/SFA.DAS.EmployerApprenticeshipsService.Web.UnitTests/ViewModels/WhenCreatedApprenticeshipListItemGroupViewModel.cs]: typo StarDatet (and line below), I think you could also remove quotes from the second param of each TestCase.

File: src/SFA.DAS.EmployerApprenticeshipsService.Web.UnitTests/TestHelper.cs (hunk 2 above)
Comment (lazcool) [reply src/SFA.DAS.EmployerApprenticeshipsService.Web.UnitTests/TestHelper.cs]: Personally, I prefer the standard Asserts over FluentAssertions.

The BeEquivalentTo method in FluentAssertions compares the elements using their Object.Equals method. That means that (in general)
1) you have to implement the Equals and GetHashCode methods on the element's type especially for the BeEquivalentTo call (if they are not already implemented)
2) it doesn't handle the case where the types are different but comparable

In this particular case, the types are KeyValuePair structs, so BeEquivalentTo would work fine.
When looking at the test, I did notice that the Assert.IsTrue was missing, so I've added that.

**Pull request merged**